documentation/Generate-code-from-TypeSpec.md

```diff
@@ -0,0 +1,116 @@
+
+Please make sure `service-dir`, `package-dir`, `packageDetails`, `isModularLibrary`, `generateMetadata`, `flavor`(for typespec-ts) is correctly configured. `experimentalExtensibleEnums`, `enableOperationGroup`, `hierarchyClient` are the optional configs.
+If you want to enable sample generation with typespec-ts, you should add 
+```
+generateSample:true
+```
+in your tspconfig.yaml
+
+
+- "parameters.service-dir.default" would be `sdk/<service>`
+- "options.@azure-tools/typespec-ts.package-dir" would be `<module>`
+
+SDK module would be generated under the SDK project folder at `sdk/<service>/<module>`.
+
+### Generate Code with code-gen-pipeline tool (recommend)
+**Notice** These steps are to generate code using the local spec repo. If you want to generate code with the github url, please refer [Generate Code with tsp-client tool](#Generate-Code-with-tsp-client-tool)
+
+Install dependencies to use code-gen-pipeline,  
+```ps
+npm install -g @azure-tools/typespec-client-generator-cli@0.14.1
+npm install -g @microsoft/rush@5.92.0
+npm install -g @azure-tools/js-sdk-release-tools
+```
+
+Create a local json file named generatedInput.json with content similar to that shown below
+```
+  {
+    "dryRun": false,
+    "specFolder": <your-local-spec-repo-path>,
+    "headSha": <commit-id-you-want-to-generate-from>,
+    "repoHttpsUrl": "https://github.com/Azure/azure-rest-api-specs",
+    "relatedTypeSpecProjectFolder": [
+    "specification/SERVICE_DIRECTORY_NAME/PACKAGE_DIRECTORY_NAME/"
+    ]
+  }
+```
+
+Run the command
+```
+code-gen-pipeline --inputJsonPath=<path-to-generatedInput.json> --outputJsonPath=<path-to-generatedOutput.json> --typespecEmitter=@azure-tools/typespec-ts --local
+```
+
+> path-to-generatedOutput.json is the detailed information of generated package, you can ignore it without pipeline. [generateOutput.json](https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/sdkautomation/GenerateOutputSchema.json) is to show us the location of generated artifact and any other messages.
+
+This command will automatically:
+1. Generate package code with TypeSpec.
+2. Build the package.
+3. Generate and run tests (optional, with warnings displayed if they fail).
+4. Generate samples, if enabled.
+5. Create or update the `CHANGELOG.md`.
+6. Bump the version according to the Azure SDK for JS policy.
+7. If this is a new package, add it to the project items in `rush.json`.
+8. Generate or update `ci.mgmt.yml` or `ci.yml` (if the package is new).
+
+
+### Generate Code with tsp-client tool
+> To reduce workload and unnecessary mistakes, it is recommended to use the simple method from the previous section. Only if you are clear about what you are doing and the method from the previous section does not meet your needs, should you consider using the method below.
+
+Install `tsp-client` CLI tool
+
+```ps
+npm install -g @azure-tools/typespec-client-generator-cli
```

Comment:
I heard we shall not use the latest version of tsp-client,  is it also true here?